### ISSUE TITLE
Several security issues in @wdio/devtools-service (xmldom)

### ISSUE THREAD

**Issue by JordiSAGE** [webdriverio/webdriverio#7433]
Several security issues in @wdio/devtools-service (xmldom)

Environment:
•	WebdriverIO version: 6.12.1
•	Mode: WDIO Testrunner (sync)
•	Node.js version: 12.19.0
•	NPM version: 6.14.8
•	Browser name and version: Chrome 91
•	Platform name and version: Windows 10
•	Additional wdio packages used (if applicable): @wdio/devtools-service 6.12.1

**Config of WebdriverIO**
exports.config = {
    runner: 'local',
    specs: 'test.feature',
    maxInstances: 1,
    capabilities: [
        {
            maxInstances: 1,
            browserName: 'chrome',
        },
    ],
    sync: true,
    deprecationWarnings: false,
    logLevel: 'warn',
    coloredLogs: false,
    screenshotPath: cfgScreenshotPath,
    bail: 0,
    baseUrl: cfgBaseUrl,
    waitforTimeout: cfgTimeout,
    connectionRetryTimeout: cfgTimeout,
    connectionRetryCount: cfgConnectionRetryCount,
    services: ['devtools', 'shared-store'],
    framework: 'cucumber',
    reporters: [
        'dot',
        'spec',
        [
            'allure',
            {
                outputDir: path.resolve(cfgReportDir + '/allure-results'),
                disableWebdriverStepsReporting: true,
                useCucumberStepReporter: true,
            },
        ],
    ],
    cucumberOpts: {
        require: cfgStepDefinitions,
        backtrace: true, 
        compiler: ['ts:ts-node/register', 'js:babel-register'],
        requireModule: [
            'tsconfig-paths/register',
            () => {
                require('ts-node').register({ files: true });
            },
        ],
        dryRun: false,
        failFast: cfgfailFast,
        format: ['pretty'],
        colors: true,
        snippets: true,
        source: true,
        profile: [],
        strict: true,
        tagsInTitle: false,
        timeout: cfgTimeout,
        ignoreUndefinedDefinitions: false,
        tagExpression: cucumberTags,
    },
    onPrepare: (config, capabilities) => {
        // Log details moved here to avoid duplication in every thread
        console.log('Features path: ' + featuresPath);
        console.log('Cucumber tags: ' + cucumberTags);
        console.log('Feature to run: ' + featureToRun);

        config.specs.forEach(featureFile => {
            console.log(featureFile);
        });
        rimraf.sync(path.resolve(cfgReportDir + '/allure-results'));
    },
    before: () => {
        require('bytenode');
        browser.setTimeout({ implicit: cfgTimeout });

        /**
         * Setup the Chai assertion framework
         */
        const chai = require('chai');

        global.expect = chai.expect;
        global.assert = chai.assert;
        global.should = chai.should();

        const puppeteer = browser.getPuppeteer();
        const page = browser.call(() => puppeteer.pages())[0];
        browser.call(() =>
            page._client.send('Page.setDownloadBehavior', {
                behavior: 'allow',
                downloadPath: process.env.downloadFolder,
            }),
        );
    },
    beforeSuite: () => {},
    beforeScenario: scenario => {},
    afterScenario: function (uri, feature, scenario, result, sourceLocation) {
        var featureName = uri.toString();
        var pathSeparator;
        if (featureName.includes('\\')) {
            pathSeparator = '\\';
        } else {
            pathSeparator = '/';
        }
        featureName = featureName.substring(featureName.lastIndexOf(pathSeparator) + 1);
        allureReporter.addArgument('Feature', featureName);
        if (result.status === 'failed') {
            browser.takeScreenshot();
        }
    },
    afterStep: function (uri, feature, { error, result, duration, passed }, stepData, context) {
        if (error !== undefined) {
            browser.takeScreenshot();
        }
        browser.pause(200);
    },
    afterFeature: suite => {},
    after: (result, capabilities, specs) => {
        createEnvironmentProperties();
    },
    onComplete: (exitCode, config) => {},
};

**Describe the bug:**
As in a previous bug reported (https://github.com/webdriverio/webdriverio/issues/7294), we have still some security issues found as part of our security checking process, in this case related again with “@wdio/devtools-service” version 6.8.2 (as well as latest version 6 6.12.1), that is getting use of the lighthouse package (lighthouse@6.5.0), that has in turn the jsonld@1.8.1 and xmldom@0.1.19 packages as dependencies, being this last package (xmldom@0.1.19) the one that produced the following security issues, detected by ThreadFix tool as Medium (Moderate on npm audit command) vulnerabilities:

Reference 1769 (View)
Component xmldom:0.1.19
Description ### Impact xmldom versions 0.6.0 and older do not correctly escape special characters when serializing elements removed from their ancestor. This may lead to unexpected syntactic changes during XML processing in some downstream applications. ### Patches Update to 0.7.0 (see issue #271 for the status of publishing the version to npm or join for Q&A/discussion #270 until it's resolved) ### Workarounds Downstream applications can validate the input and reject the maliciously crafted documents. ### For more information If you have any questions or comments about this advisory: * Open an issue in [`xmldom/xmldom`](https://github.com/xmldom/xmldom) * Email us: send an email to **all** addresses that are shown by `npm owner ls xmldom`
http://cve.mitre.org/cgi-bin/cvename.cgi?name=1769

Reference CVE-2021-21366 (View)
Component xmldom:0.1.19
Description xmldom is a pure JavaScript W3C standard-based (XML DOM Level 2 Core) DOMParser and XMLSerializer module. xmldom versions 0.4.0 and older do not correctly preserve system identifiers, FPIs or namespaces when repeatedly parsing and serializing maliciously crafted documents. This may lead to unexpected syntactic changes during XML processing in some downstream applications. This is fixed in version 0.5.0. As a workaround downstream applications can validate the input and reject the maliciously crafted documents.
http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-21366

Reference CVE-2021-32796 (View)
Component xmldom:0.1.19
Description xmldom is an open source pure JavaScript W3C standard-based (XML DOM Level 2 Core) DOMParser and XMLSerializer module. xmldom versions 0.6.0 and older do not correctly escape special characters when serializing elements removed from their ancestor. This may lead to unexpected syntactic changes during XML processing in some downstream applications. This issue has been resolved in version 0.7.0. As a workaround downstream applications can validate the input and reject the maliciously crafted documents.
http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-32796

xmldom:0.1.19 dependency affected:
`-- @wdio/devtools-service@6.12.1
  `-- lighthouse@6.5.0
    `-- jsonld@1.8.1
      `-- **xmldom@0.1.19** 

**Due to our requeriments to be compliant with other applications we manage, it would be required to base the fix or to generate a new issue version based in webdriverio version 6 (We're currently using WebdriverIO version 6.8.2, but we could change to version 6.12.1 or above if needed)**

**To Reproduce**
Check on ThreadFix security vulnerability platform or npm audit command.

**Expected behavior**
@wdio/devtools-service new version based on version 6, should include lighthouse package above 6.5.0 that in turn has to incorporate a jsonld version with xmldom package >=0.7.0

**Additional context**
As commented in the previous bug reported, we're close to a new release date and this issue is stopping us to provide a new version, Thanks in advance for your understanding.

**Comment by christian-bromann** [comment]
Feel free to provide a PR with a patch to update lighthouse in the `v6` branch. I am happy to make a release. Otherwise I recommend to update to v7 of WebdriverIO.
